### PR TITLE
Bugfix/rm 8853 boc boolean value boc enum value and boc list use incorrect elements as target for the label for attribute

### PR DESCRIPTION
**Pull request by Frazerus**
https://re-motion.atlassian.net/browse/RM-8853

Implementation for the non hover issues

### CODE REVIEW

File: Remotion/ObjectBinding/Web/UI/Controls/BocEnumValue.cs [hunk 1]
```diff
@@ -445,7 +445,15 @@ public override Control TargetControl
     [Browsable(false)]
     public string? FocusID
     {
-      get { return IsReadOnly ? null : GetValueName(); }
+      get
+      {
+        if (IsReadOnly)
+          return null;
+        if (ListControlStyle.ControlType == ListControlType.RadioButtonList)
+          return Controls.Count > 0 ? GetValueName() + "_0" : null;
```
Comment (MichaelKetting) [inline Remotion/ObjectBinding/Web/UI/Controls/BocEnumValue.cs:453]:
```suggestion
        if (IsReadOnly)
          return null;
        if (ListControlStyle.ControlType == ListControlType.RadioButtonList)
          return Controls.Count > 0 ? GetValueName() + "_0" : null;
```
```suggestion
        if (IsReadOnly)
          return null;

        if (ListControlStyle.ControlType == ListControlType.RadioButtonList)
          return Controls.Count > 0 ? GetValueName() + "_0" : null;
```

Comment (MichaelKetting) [inline Remotion/ObjectBinding/Web/UI/Controls/BocEnumValue.cs:453]: Please add a comment that the ID is calculated by System.Web.U.ListBox. Some day I want to rewrite the rendering without relying on ASP.NET WebForms, then we could conceivably remove this duplication.

File: Remotion/Web/Core/UI/Controls/SmartLabel.cs [hunk 2]
```diff
@@ -184,12 +184,15 @@ protected virtual void LoadResources (IResourceManager resourceManager)
...
-      if (((ISmartControl)target).UseLabel)
-        return ((IFocusableControl)target).FocusID;
+      if (!smartTarget.UseLabel)
+        return null;
+
+      if (target is IFocusableControl focusableControl)
+        return focusableControl.FocusID;
 
-      return null;
+      return target.ClientID;
```
Comment (MichaelKetting) [inline Remotion/Web/Core/UI/Controls/SmartLabel.cs:195]: I updated the docs in https://re-motion.atlassian.net/browse/RM-8853 for the "best" implementation. I think it already covers most of the notes but I added a case for IFocusableControl without ISmartControl. Please verify the logic against the updated spec and check if you want to extract this into a separate bug. If you see something strange, please question :)

**Review by MichaelKetting — COMMENTED** [review]
Looks good so far. Please double-check against amended spec in https://re-motion.atlassian.net/browse/RM-8853

**Comment by Frazerus** [comment]
### No hover:
image

### Old hover effect:
Hover over label:
image
Hover over specific item:
image

### Trial hover effect on bocEnumValue
Hover over label:
image
Hover over specific item:
image
Hover over selected item:
image

File: Remotion/Web/Core/Res/Themes/NovaViso/HTML/Common.css [hunk 3]
```diff
@@ -276,6 +284,8 @@ input[type="password"].remotion-themed,
   background-image: url('../Image/sprite.svg#CheckBoxFalse');
 }
 
+.remotion-themed input[type=checkbox]:not(:disabled):hover + label::before,
+.remotion-themed input[type=checkbox]:not(:disabled):hover + span::before,
```
Comment (MichaelKetting) [inline Remotion/Web/Core/Res/Themes/NovaViso/HTML/Common.css:288]: A few lines up we have 
```
.remotion-themed input[type=checkbox] + label:not(:hover)::before,
.remotion-themed input[type=checkbox] + span:not(:hover)::before,
```

I think are should hadd the not-hover rule too.
```
.remotion-themed input[type=checkbox]:not(:hover) + label::before,
.remotion-themed input[type=checkbox]:not(:hover) + span::before,
```
Please check this for the other added rules, too. 

The not-hover rules are added to make it explicit that it's not-hover / the default scenario, so while it's possible that there is no actual effect, having the rules symetrical makes the reasoning easier to follow. OR there is actually a selector-priority-based reason why we need them.

**Review by MichaelKetting — COMMENTED** [review]
partial review. some fixed to come :)

File: Remotion/Web/Core/Res/Themes/NovaViso/HTML/Common.css (hunk 3 above)
Comment (MichaelKetting) [reply Remotion/Web/Core/Res/Themes/NovaViso/HTML/Common.css:290]: Note: already pushed as a new commit to add missing stuff.

File: Remotion/Web/Core/Res/Themes/NovaViso/HTML/Common.css [hunk 4]
```diff
@@ -19,6 +19,7 @@
   --color-primary-highlight: #218FC9;
   --color-primary-text: #FFFFFF;
   --color-highlight: #DEF3FE;
+  --color-sibling-hover: #f3faff;
```
Comment (MichaelKetting) [inline Remotion/Web/Core/Res/Themes/NovaViso/HTML/Common.css:22]: Considering the set of colors was created based on a color palette and design, I prefer not to add new colors, in particular colors used only once. Am pushing a version where the entire background is highlighted when hovering the label and only the respective OPTION is highlighted when hovering the ELEMENT element.

**Review by MichaelKetting — CHANGES_REQUESTED** [review]
I tweaked the CSS. please counter-review.
Some of the review-comments are already completed and only left open for reviewing.

File: Remotion/Web/Core/Res/Themes/NovaViso/HTML/Common.css (hunk 4 above)
Comment (MichaelKetting) [reply Remotion/Web/Core/Res/Themes/NovaViso/HTML/Common.css]: Comment completed. Please review.

File: Remotion/Web/Core/Res/Themes/NovaViso/HTML/Common.css (hunk 3 above)
Comment (MichaelKetting) [reply Remotion/Web/Core/Res/Themes/NovaViso/HTML/Common.css:292]: Comment completed. please review.

File: Remotion/Web/Core/Res/Themes/NovaViso/HTML/Common.css (hunk 4 above)
Comment (Frazerus) [reply Remotion/Web/Core/Res/Themes/NovaViso/HTML/Common.css]: Looks good

File: Remotion/ObjectBinding/Web.ClientScript/Scripts/BocBooleanValue.ts [hunk 5]
```diff
@@ -106,13 +106,10 @@ class BocBooleanValue_Resource
...
 
-
     checkboxSpan.setAttribute('aria-checked', checkedState!);
 
-    const isHover = icon.src == icon.dataset['srcHover'];
-    icon.dataset["src"] = iconSrc!;
-    icon.dataset['srcHover'] = iconHoverSrc!;
-    icon.src = isHover ? icon.dataset["src"] : icon.dataset['srcHover'];
+    icon.style.setProperty("--bocbooleanvalue-standard-background", "url('" + iconSrc! + "')");
+    icon.style.setProperty("--bocbooleanvalue-hover-background", "url('" + iconHoverSrc! + "')");
```
[the remark this answers is not in the document]
Comment (Frazerus) [reply Remotion/ObjectBinding/Web.ClientScript/Scripts/BocBooleanValue.ts:112]: In that case, we would be unable to use the resouce manager thingy to change the urls, we need to set css urls programmatically.

File: Remotion/ObjectBinding/Web/Res/Themes/NovaGray/Html/BocBooleanValue.css
Comment (MichaelKetting) [inline Remotion/ObjectBinding/Web/Res/Themes/NovaGray/Html/BocBooleanValue.css:1]: We don't need hover in NovaGray :)

**Review by MichaelKetting — COMMENTED** [review]
There's 19 or so red tests left so I'll wait with the approval until they're done.

**Review by MichaelKetting — APPROVED** [review]

**Pull request merged**